src/main/scala/viper/silver/parser/Resolver.scala

```diff
@@ -293,6 +236,46 @@ case class TypeChecker(names: NameAnalyser) {
     }
   }
 
+  def checkAssign(stmt: PAssign): Unit = {
+    // Check targets
+    stmt.targets foreach {
+      case idnuse: PIdnUse => names.definition(curMember)(idnuse) match {
+          case Some(decl: PAssignableVarDecl) =>
+            check(idnuse, decl.typ)
+          case _ =>
+            messages ++= FastMessaging.message(idnuse, "expected an assignable identifier as lhs")
+        }
+      case fa@PFieldAccess(_, field) => names.definition(curMember)(field, Some(PField.getClass)) match {
+          case Some(PFieldDecl(_, typ)) =>
+            check(fa, typ)
+          case _ =>
+            messages ++= FastMessaging.message(field, "expected a field as lhs")
+        }
+      case call: PCall => sys.error(s"Unexpected node $call found")
+    }
+    // Check rhs
+    stmt match {
+      case PAssign(targets, PCall(func, args, _)) if names.definition(curMember)(func).get.isInstanceOf[PMethod] =>
+        val PMethod(_, formalArgs, formalTargets, _, _, _) = names.definition(curMember)(func).get.asInstanceOf[PMethod]
+        formalArgs.foreach(fa => check(fa.typ))
+        if (formalArgs.length != args.length) {
+          messages ++= FastMessaging.message(stmt, "wrong number of arguments")
+        } else if (formalTargets.length != targets.length) {
+            messages ++= FastMessaging.message(stmt, "wrong number of targets")
+        } else {
+          for ((formal, actual) <- (formalArgs zip args) ++ (formalTargets zip targets)) {
+            check(actual, formal.typ)
+          }
+        }
+      case PAssign(Seq(target), PNewExp(fieldsOpt)) =>
+        check(target, Ref)
+        fieldsOpt map (acceptAndCheckTypedEntity[PFieldDecl, Nothing](_, "expected a field as argument"))
+      case PAssign(Seq(lhs), rhs) => check(rhs, lhs.typ)
```

Comment:
Wait. So I can now do
``var i: Int, j: Int := e``
and that will assign e to all the newly-declared variables, but I can't write
``i, j := e``,
right?
But I can write 
``i, j := myMethod()``
if the method returns two results, but I cannot write
``var i: Int, j: Int := myMethod()``
if the method returns two results? But maybe I can if it returns only one result?

Reply:
No, that's not the intended behaviour. The highlighted case requires `Seq(lhs)` (i.e. exactly one target), thus both (1) and (2) are invalid and both (3) and (4) are valid. We could consider making (1) valid, but it's not really clear if it should assign `e` to all new vars or just the `j`. Imo we should just keep it as is and reject (1).

Reply:
Okay, yes, that makes sense.